docs/docsite/rst/dev_guide/developing_program_flow_modules.rst

```diff
@@ -5,42 +5,37 @@
 Ansible module architecture
 ***************************
 
-This in-depth dive helps you understand Ansible's program flow to execute
-modules. It is written for people working on the portions of the Core Ansible
-Engine that execute a module. Those writing Ansible Modules may also find this
-in-depth dive to be of interest, but individuals simply using Ansible Modules
-will not likely find this to be helpful.
+If you're working on Ansible's Core code, or writing an Ansible module, this deep dive helps you understand how Ansible's program flow executes
+modules. If you're just using Ansible Modules in playbooks, you can skip this section.
 
-.. contents:: Topics
+.. contents::
    :local:
 
 .. _flow_types_of_modules:
 
 Types of modules
 ================
 
-Ansible supports several different types of modules in its code base.  Some of
+Ansible supports several different types of modules in its code base. Some of
 these are for backwards compatibility and others are to enable flexibility.
 
 .. _flow_action_plugins:
 
 Action plugins
 --------------
 
-Action Plugins look like modules to end users who are writing :term:`playbooks` but
-they're distinct entities for the purposes of this document.  Action Plugins
-always execute on the controller and are sometimes able to do all work there
-(for instance, the ``debug`` Action Plugin which prints some text for the user to
-see or the ``assert`` Action Plugin which can test whether several values in
-a playbook satisfy certain criteria.)
-
-More often, Action Plugins set up some values on the controller, then invoke an
-actual module on the managed node that does something with these values.  An
-easy to understand version of this is the :ref:`template Action Plugin
-<template_module>`.  The :ref:`template Action Plugin <template_module>` takes values from
+Action plugins look like modules to anyone writing a playbook. They are documented as modules. However, they behave differently on the back end. Action plugins always execute on the controller.
```

Comment:
'they are documented as modules' can be read in 2 ways, 'their usage documentation lives inside a module' or 'the documentation makes no distinction on module vs action plugin', the latter being not completely true since we do reflect that 'this module has an action plugin' and that they are diff types than modules. 

I would rephrase:
"The 'action' of a task is normally assumed to be the same as a module of the same name, but there is always an 'action plugin' involved, sometimes they just execute the  module, other times they do all the work and the module is just there as documentation, in other cases it is something in between."

"Then the paragraph below is good with examples, i would also add "then modules that don't have a corresponding action plugin are executed by the `<link normal section below>`'normal' action plugin`</link>`"